include/fiction/layouts/coordinates.hpp

```diff
@@ -832,10 +728,132 @@ constexpr coord_t to_siqad_coord(const CoordinateType& coord) noexcept
 
 }  // namespace siqad
 
+/**
+ * An iterator type that allows to enumerate coordinates in order within a boundary.
+ *
+ * @tparam CoordinateType Type of coordinate to enumerate.
+ */
+template <typename CoordinateType>
+class coord_iterator
+{
+  public:
+    using value_type = CoordinateType;
+    /**
+     * Standard constructor. Initializes the iterator with a starting position and the boundary within to enumerate.
+     *
+     * With `dimension = (1, 2, 1)` and `start = (0, 0, 0)`, the following order would be enumerated:
+     *
+     * - (0, 0, 0)
+     * - (1, 0, 0)
+     * - (0, 1, 0)
+     * - (1, 1, 0)
+     * - (0, 2, 0)
+     * - (1, 2, 0)
+     * - (0, 0, 1)
+     * - (1, 0, 1)
+     * - (0, 1, 1)
+     * - (1, 1, 1)
+     * - (0, 2, 1)
+     * - (1, 2, 1)
+     *
+     * coord_iterator is compatible with the STL forward_iterator category.
+     *
+     * @param dimension Boundary within to enumerate. Iteration wraps at its limits.
+     * @param start Starting coordinate to enumerate first.
+     */
+    constexpr explicit coord_iterator(const CoordinateType& dimension, const CoordinateType& start) noexcept :
+            aspect_ratio{dimension},
+            coord{start}
+    {}
+    /**
+     * Increments the iterator.
+     *
+     * @return Reference to the incremented iterator.
+     */
+    constexpr coord_iterator& operator++() noexcept
+    {
+        if (coord != aspect_ratio)
+        {
+            ++coord.x;
+
+            if (coord.x > aspect_ratio.x)
+            {
+                coord.x = 0;
+
+                if constexpr (std::is_same_v<CoordinateType, offset::ucoord_t>)
+                {
+                    ++coord.y;
+                    if (coord.y > aspect_ratio.y)
+                    {
+                        coord.y = 0;
+
+                        ++coord.z;
+                    }
+                }
+                else if constexpr (std::is_same_v<CoordinateType, siqad::coord_t>)
+                {
+                    coord.y += coord.z;
+                    coord.z = !coord.z;
+                }
+                else
```

Comment:
How about `cube::coord_t`?

Reply:
Same as offset::ucoord_t, implemented in the latest commit